moveos/moveos-stdlib/moveos-stdlib/sources/context.move

```diff
@@ -94,18 +94,44 @@ module moveos_std::context {
     }
 
 
-    // Wrap functions for StorageContext
+    // Wrap functions for Object
+
+    #[private_generics(T)]
+    /// Create a new Object, Add the Object to the global object storage and return the Object
+    /// Note: the default owner is the `System`, the caller should explicitly transfer the Object to the owner.
+    /// The owner can get the `&mut Object` by `borrow_mut_object`
+    public fun new_object<T: key>(self: &mut Context, value: T): Object<T> {
+        let id = fresh_object_id(self);
+        object::new(id, value)
+    }
+
+    public(friend) fun new_object_with_id<T: key>(_self: &mut Context, id: ObjectID, value: T) : Object<T> {
+        object::new(id, value)
+    }
+
+    #[private_generics(T)]
+    /// Create a new singleton object, the object is owned by `System` by default.
+    /// Singleton object means the object of `T` is only one instance in the Object Storage.
+    public fun new_singleton<T: key>(_self: &mut Context, value: T): Object<T> {
+        object::new_singleton(value)
+    }
 
     /// Borrow Object from object store with object_id
-    /// Any one can borrow an &Object from the global object storage
+    /// Any one can borrow an `&Object<T>` from the global object storage
     public fun borrow_object<T: key>(_self: &Context, object_id: ObjectID): &Object<T> {
         let object_entity = object::borrow_from_global<T>(object_id);
         object::as_ref(object_entity)
     }
 
+    /// Borrow singleton Object from global object storage
+    public fun borrow_singleton<T: key>(self: &Context): &Object<T> {
+        let object_id = object::singleton_object_id<T>();
+        borrow_object(self, object_id)
+    }
+
     /// Borrow mut Object from object store with object_id
-    /// If the object is not shared, only the owner can borrow an &mut Object from the global object storage
-    public fun borrow_object_mut<T: key>(_self: &mut Context, owner: &signer, object_id: ObjectID): &mut Object<T> {
+    /// If the object is not shared, only the owner can borrow an `&mut Object<T>` from the global object storage
+    public fun borrow_mut_object<T: key>(_self: &mut Context, owner: &signer, object_id: ObjectID): &mut Object<T> {
```

Comment:
`borrow_xxx_mut` and `borrow_mut_xxx` can be unified in rooch framework

Reply:
Only one `borrow_box_mut` in the raw_table is an private native function. We can refactor it in the future.